enginetest/queries/insert_queries.go

```diff
@@ -1606,6 +1606,76 @@ var InsertScripts = []ScriptTest{
 			},
 		},
 	},
+	{
+		Name: "insert on duplicate key for keyless table",
+		SetUpScript: []string{
+			`create table t (i int unique, j varchar(128))`,
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query: `insert into t values (0, "first")`,
+				Expected: []sql.Row{
+					{types.NewOkResult(1)},
+				},
+			},
+			{
+				Query: `insert into t values (0, "second") on duplicate key update j = "third"`,
+				Expected: []sql.Row{
+					{types.NewOkResult(2)},
+				},
+			},
+			{
+				Query: `select i, j from t order by i`,
+				Expected: []sql.Row{
+					{0, "third"},
+				},
+			},
+		},
+	},
+	{
+		Name: "insert on duplicate key for keyless table multiple unique columns",
```

Comment:
Only additional tests I can think of is maybe something like `insert into t values (0,0,0), (0,0,0),(0,0,1),(0,0,1) on duplicate key update c3 = 1`. And then a variation with nulls maybe? `insert into t values (0,null,0), (0,null,0) on duplicate key update c2 = null/0`

Reply:
sometimes ordinals cause problems, maybe one with `unique key (c3,c2,c1)` could trigger some edge case

Reply:
The tests with nulls and ordinals works, so I added those.
The batched inserts do not work in GMS, but work fine in dolt, so I added a skipped test for that.
I have a fix for that, but it breaks other stuff, so I think a follow up PR and a new tracking would be better.